File: Xylophone_ObjC/MainViewController.m
#import "MainViewController.h"
#import <AVFoundation/AVFoundation.h>

@interface MainViewController ()
@property (nonatomic) AVAudioPlayer* audioPlayer;
@end

@implementation MainViewController

-(IBAction) notePressed:(UIButton*)sender
{
    NSString* soundFileName = [NSString stringWithFormat:@"%@%ld",@"note",[sender tag]];
    [self playSoundWithFile:soundFileName];
}

-(void) playSoundWithFile: (NSString*) name
{
    NSURL* url = [[NSBundle mainBundle] URLForResource: name withExtension:@"wav"];
    
    @try
    {
        _audioPlayer = [[AVAudioPlayer alloc] initWithContentsOfURL:url error:nil];
        if([self audioPlayer] == nil)
        {
            return;
        }
        [[self audioPlayer] prepareToPlay];
        [[self audioPlayer] play];
    }
    @catch(NSException *exception)
    {
        NSLog(@"AVAudio Player could not play the soud due to: %@",[exception description]);
    }
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self setupNavigation];
}

- (void) setupNavigation
{
    if (@available(iOS 11.0, *))
    {
        [[[self navigationController] navigationBar] setPrefersLargeTitles:YES];
    }
    [self setTitle: @"XylophoneObjC"];
    [[[self navigationController] navigationBar] setBarStyle:UIBarStyleBlackTranslucent];
    [[[self navigationController] navigationBar] setTintColor:[UIColor whiteColor]];
}
@end

File: Xylophone_Swift/MainViewController.swift
import UIKit
import AVFoundation

@objc(MainViewController)
class MainViewController: UIViewController {
    
    @IBAction
    func notePressed(_ sender: UIButton) {
        playSoundForFileWith(name: "note\(sender.tag)")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        setupNavigation()
    }

    private func setupNavigation(){
        if #available(iOS 11.0, *) {
            navigationController?.navigationBar.prefersLargeTitles = true
        }
        title = "XylophoneSwift"
        navigationController?.navigationBar.barStyle = .blackTranslucent
        navigationController?.navigationBar.tintColor = .white
    }
    
    private func playSoundForFileWith(name: String) {
        let url = Bundle.main.url(forResource: name, withExtension: "wav")!
        do {
            audioPlayer = try AVAudioPlayer(contentsOf: url)
            guard let player = audioPlayer else { return }
            player.prepareToPlay()
            player.play()
        }catch {
            print(error.localizedDescription)
        }
    }
    
    private var audioPlayer: AVAudioPlayer!
}

